hwtypes/modifiers.py

```diff
@@ -128,3 +131,46 @@ def wrap_modifier(T, mods):
     for mod in mods:
         wrapped = mod(wrapped)
     return wrapped
+
+#Takes an ADT type and removes any modifiers on any level of the Tree
+def strip_modifiers(adt_t):
+    #remove modifiers from this level
+    adt_t, _ = unwrap_modifier(adt_t)
+    if (issubclass(adt_t, AbstractBit) or issubclass(adt_t, AbstractBitVector)):
+        return adt_t
+    elif issubclass(adt_t, Enum):
+        return adt_t
+    elif issubclass(adt_t, Sum):
```

Comment:
Check `issubclass(adt_t, GetitemSyntax)`, for Sum/Tuple, build new type with `adt_t.unbound_t[new_fields]`
Check `issubclass(adt_t, AttrSyntax) and adt.FIELD_T is type` for Product/TaggedUnion and exclude Enum.  Build new type similarly to above.
else return adt_t

Reply:
@cdonovick do you mean issubclass(type(adt_t), AttrSyntax) ?

Reply:
I meant `isinstance`